community/server/src/main/java/org/neo4j/server/configuration/ConfigurationBuilder.java

```diff
@@ -137,4 +141,13 @@ public Configuration configuration()
         }
 
     }
+
+    class DefaultSettingsClassesHolder
```

Comment:
Not really sure that i fully understand the question :), The idea of that class to provide common place where different configurations can get settings classes